spec/support/oauth_helpers.rb

```diff
@@ -35,4 +35,13 @@ def token_refresh_params(refresh_token)
       refresh_token: refresh_token
     }
   end
+
+  def create_token_header(user = nil)
+    user ||= Fabricate(:user)
+
+    application = Fabricate(:application)
+    access_token = Fabricate(:access_token, resource_owner_id: user.id, application_id: application.id)
+
+    [{ 'Authorization' => "Bearer #{access_token.token}" }, user]
+  end
```

Comment:
I'm not sure if my suggestion works, but it would be nice if we could use this helper like `login_as user` instead of creating the token and parring them manually to the request each time 🤔 

```suggestion
  def login_as(user = nil)
    user ||= Fabricate(:user)

    application = Fabricate(:application)
    access_token = Fabricate(:access_token, resource_owner_id: user.id, application_id: application.id)

    request.headers['Authorization'] = "Bearer #{access_token.token}"
    
    user
  end
```

I let you try~ 🙌

Reply:
@malparty I also would like this to work as well but from limited understanding, `request` is not accessible at this point for this version of RSpec? Not sure is there any import I am missing.

Reply:
From my understanding, the Request spec cannot get access to the request and request.header, we have to use [Controller spec](https://rspec.info/features/6-0/rspec-rails/controller-specs/) instead. But the Controller spec does not seem to be made for API testing like we are doing. Also I found this section

> We encourage you to use [request specs](https://rspec.info/features/6-0/rspec-rails/controller-specs/request-specs/request-spec) if you want to set headers in your call. If you still want to use controller specs with custom http headers you can use request.headers:

Reply:
Oh, I see. 
The `request` variable is actually present and reachable. The problem is that the variable is `nil` because the test did not perform any request yet.

There are two options then:
- A) either keep the current implementation 
- B) either replace the token fabrication with a call to the login API so that it initialize the request object and provide you in the response the required token

Both look fine. A) is probably farter to execute // B) looks a bit more standard 😇

Reply:
Let me try `B`, not sure if the `request` variable from the login call and the call from the spec will be the same.

Reply:
I updated the call for `create_token_header` to make it more streamline for the call-side. Since we are passing `user?` object we don't have to return it if the spec requires a user check.

I was not able to do `B` because the `request` from the first API call is not used in the following API calls.

Reply:
That's strange. Maybe it works a bit differently in a web API (in Flexi we just pass the devise headers to the request instance and it would work for all calls. Or maybe the helper from Devise is doing additional things 🙈 ).

Let's keep your current implementation then :)